compiler/modules/types/core.bal

```diff
@@ -171,7 +172,10 @@ function cellAtomType(Atom atom) returns CellAtomicType {
 }
 
 // See memoSubtypeIsEmpty for what these mean.
-type MemoEmpty boolean|"provisional"|();
+// loop means type has a recursive shape
```

Comment:
These are not adequate explanations of the meaning of the values.

Reply:
fixed